### PR TITLE
Graphile auth and UI

### PR DESCRIPTION
**Pull request by german-mergestat**
Implementing login an logout functionalities.

- It is missing unit testing

login

### CODE REVIEW

File: ui/src/api-logic/axios/api.ts [hunk 1]
```diff
@@ -17,3 +17,31 @@ export const validateGtihubToken = async (pat: string) => {
...
+ * @param user Database user
+ * @param password Database password
+ * @returns true if it is logged in, otherwise returns false
+ */
+export const auth = async (user: string, password: string) => {
+  try {
+    const response = await axios.post('/api/admin-auth', { user, password })
+    return response.status === 200
+  } catch (error) {
+    return false
```
Comment (patrickdevivo) [inline ui/src/api-logic/axios/api.ts:32]: we should handle the `error` here in some way, even if it's just logging it out to the console for now (maybe display in an alert if we can?)

just so we dont fail silently for now

File: ui/src/api-logic/axios/api.ts [hunk 2]
```diff
@@ -17,3 +17,31 @@ export const validateGtihubToken = async (pat: string) => {
     return false
   }
 }
+
+/**
+ * Method to login to Postgrahile
```
Comment (patrickdevivo) [inline ui/src/api-logic/axios/api.ts:22]:
```suggestion
 * Method to login to Postgraphile
```

File: ui/src/api-logic/axios/api.ts (hunk 1 above)
Comment (german-mergestat) [reply ui/src/api-logic/axios/api.ts]: Done [65cc09f](https://github.com/mergestat/fuse/pull/334/commits/65cc09f4eb0c07e3680c9002f5e745931c2ca6b9)

File: ui/src/api-logic/axios/api.ts (hunk 2 above)
Comment (german-mergestat) [reply ui/src/api-logic/axios/api.ts]: Done [65cc09f](https://github.com/mergestat/fuse/pull/334/commits/65cc09f4eb0c07e3680c9002f5e745931c2ca6b9)

File: ui/lib/apollo.ts [hunk 3]
```diff
@@ -1,18 +1,33 @@
...
+  uri: '/api/graphql',
+  credentials: 'same-origin', // Additional fetch() options like `credentials` or `headers`
+})
+
+const logoutLink = onError(({ graphQLErrors }) => {
+  if (graphQLErrors) {
+    for (const err of graphQLErrors) {
+      if (err.message.includes('permission denied for')) {
+        if (typeof window !== 'undefined') {
+          window.location.href = '/login?lostSession=true'
```
Comment (patrickdevivo) [inline ui/lib/apollo.ts:20]: should we use `next/router` here instead?

Comment (german-mergestat) [reply ui/lib/apollo.ts]: Done [ab457e9](https://github.com/mergestat/fuse/pull/334/commits/ab457e9d1aad3416096942ee53b5eff1d41fe9d6)

File: ui/src/pages/login/index.tsx [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+            <Icon
+              as="/logo.svg"
+              width={144}
+              className="flex w-auto items-center"
+            />
+          </Panel.Header>
+          <Panel.Body>
+            {lostSession && (
+              <Alert theme="light" type="warning" className="mb-6" >
+                You’ve lost session. Please log in again.
```
[the remark this answers is not in the document]
Comment (german-mergestat) [reply ui/src/pages/login/index.tsx]: Done [ab457e9](https://github.com/mergestat/fuse/pull/334/commits/ab457e9d1aad3416096942ee53b5eff1d41fe9d6)

**Review by patrickdevivo — APPROVED** [review]

**Pull request merged**